### PR TITLE
Event broadcast

### PR DESCRIPTION
**Pull request by tank157**
Another PR for moving signup notys to backend for NH and friends

Makes it easy to generate a reg form (and get instructions/authority) from a user and optional election.

### CODE REVIEW

File: src/voter_registration_works/handlers.clj [hunk 1]
```diff
@@ -164,7 +171,49 @@
...
 (def voter-register
-  (handle-errors voter-register*))
+  (handle-errors (partial voter-register* channels/status-create)))
+
+(defn-traced generate-registration-form*
+  "Generates a NVRF registration form for a user and optional election,
+   as they are returned by user-works and election-works. Will only generate
+   the form if the user has a `:by-mail` preference, unless the message
+   contains `:force true`."
+  [{:keys [user election force] :as message}]
```
Comment (tank157) [inline src/voter_registration_works/handlers.clj]: take another look at this

File: src/voter_registration_works/handlers.clj [hunk 2]
```diff
@@ -164,7 +171,49 @@
...
+                       "Response:" response)
+            (throw (Exception. "Error rendering NVRF form")))
+          (let [state-kw (-> user
+                             :addresses
+                             :registered
+                             :state
+                             ->lower-case-keyword)
+                form-uri (:uri response)
+                instructions (reg-form/by-mail-instructions state-kw)]
+            (if-let [authority (some-> election
```
Comment (tank157) [inline src/voter_registration_works/handlers.clj]: this probably should be electorate if it exists

File: src/voter_registration_works/states.clj [hunk 3]
```diff
@@ -46,6 +46,13 @@
   (let [ocd-id (state->ocd-id state)]
     (election-authority {:ocd-id ocd-id, :language language})))
 
+(defn-traced registration-methods->by-type
+  "Returns has here registration-methods are
+   keyed to their type"
+  [registration-methods]
+  (reduce #(merge %1 {(:type %2) (dissoc %2 :type)})
```
Comment (ddellacosta) [inline src/voter_registration_works/states.clj]: Stylistic suggestion, take it or leave it: I might find it _slightly_ more clear and concise to do

```clojure
(reduce #(assoc %1 (:type %2) (dissoc %2 :type)) {} registration-methods)
```

File: src/voter_registration_works/handlers.clj [hunk 4]
```diff
@@ -136,23 +132,28 @@
...
+(defn-traced initiate-registration-status
+  [register-request create-event-channel]
+  (let [status
+        (registration-status/upsert-registration-status
+         (dt/db)
+         {:user-id (get-in register-request [:voter :user-id])
+          :method (:method register-request)
+          :source :turbovote
+          :status :initiated})]
+    (async/put! create-event-channel status)
```
Comment (ddellacosta) [inline src/voter_registration_works/handlers.clj]: Sorry I blew past this and approved this PR already, but I'm just curious: do we use this or the delete event anywhere yet and I missed it?

Comment (tank157) [inline src/voter_registration_works/handlers.clj]: Yes, it's one of the events that will reschedule the signup notification.

Comment (ddellacosta) [inline src/voter_registration_works/handlers.clj]: Gotcha, and I just caught it on the other side in `election-notification-works` I believe. Thanks for the explanation.

**Pull request merged**